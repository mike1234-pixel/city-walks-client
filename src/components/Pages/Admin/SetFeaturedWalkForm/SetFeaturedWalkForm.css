.set-featured-walk-form {
  padding-top: 20px;
  padding-bottom: 20px;
}
