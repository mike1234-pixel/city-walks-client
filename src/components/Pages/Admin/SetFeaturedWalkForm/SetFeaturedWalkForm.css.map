{"version":3,"sourceRoot":"","sources":["SetFeaturedWalkForm.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;EACA","file":"SetFeaturedWalkForm.css"}