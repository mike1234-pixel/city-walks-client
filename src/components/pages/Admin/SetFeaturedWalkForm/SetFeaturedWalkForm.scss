.set-featured-walk-form {
  padding: 20px;
}
