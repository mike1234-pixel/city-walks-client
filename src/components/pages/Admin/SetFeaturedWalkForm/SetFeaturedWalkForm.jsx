import { useState } from "react"
import { MDBInput, MDBBtn, MDBIcon } from "mdbreact"
import axios from "axios";
import qs from "qs";
import toTitleCase from "../../../../functions/toTitleCase"
import './SetFeaturedWalkForm.css'

const SetFeaturedWalkForm = () => {
    const [featuredWalk1, setFeaturedWalk1] = useState("")
    const [featuredWalk2, setFeaturedWalk2] = useState("")
    const [featuredWalk3, setFeaturedWalk3] = useState("")

    const handleChange = (event) => {
        switch(event.target.name) {
            case "featured-walk-1":
              setFeaturedWalk1(event.target.value)
              break;
            case "featured-walk-2":
              setFeaturedWalk2(event.target.value)
              break;
            case "featured-walk-3":
              setFeaturedWalk3(event.target.value)
              break;
          } 
    }

    const handleSubmit = (event) => {
        event.preventDefault()

        let payload = {
            featuredWalk1: toTitleCase(featuredWalk1),
            featuredWalk2: toTitleCase(featuredWalk2),
            featuredWalk3: toTitleCase(featuredWalk3),
          };

        axios
        .patch("http://localhost:5000/set-featured-walk", qs.stringify(payload))
        .then((res, err) => {
          if (err) {
            console.log(err);
          } else if (res.data === "featured walk set") {
            console.log("featured walk set")
          } else {
            console.log("featured walk not set")
          }
        });
        alert("Featured walks set")
        setFeaturedWalk1("")
        setFeaturedWalk2("")
        setFeaturedWalk3("")
        window.scrollTo(0, 0);

    }
    return (
        <div>
        <h2>Set Featured Walks</h2>
        <p>Administrator note: a prerequisite of this working is that at least one record has the featuredWalk field set to true.</p>
        <p>Only records where walk is fully in Title Case can be set as a featured walk. For example, 'Bond Street to Baker Street' will not pass.</p>
        <p>The form will convert your input to Title Case.</p>
        <form onSubmit={handleSubmit} className="set-featured-walk-form">
            <MDBInput type="text" name="featured-walk-1" id="featured-walk-1" value={featuredWalk1} label="featured walk 1 (walk name, e.g 'Stoke Newington')" onChange={handleChange} required/>
            <MDBInput type="text" name="featured-walk-2" id="featured-walk-2" value={featuredWalk2} label="featured walk 2 (walk name, e.g 'Buckingham Palace')" onChange={handleChange} required/>
            <MDBInput type="text" name="featured-walk-3" id="featured-walk-3" value={featuredWalk3} label="featured walk 3 (walk name, e.g 'Old Street')" onChange={handleChange} required/>
            <MDBBtn outline color="elegant" type="submit">
                Set Featured Walks
            <MDBIcon far icon="star" className="ml-2" />
          </MDBBtn>
        </form>
    </div>
    )
}

export default SetFeaturedWalkForm