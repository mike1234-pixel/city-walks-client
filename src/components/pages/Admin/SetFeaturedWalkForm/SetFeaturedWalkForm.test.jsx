import renderer from 'react-test-renderer'
import SetFeaturedWalkForm from "./SetFeaturedWalkForm"

// snapshot test
it('SetFeaturedWalkForm Page renders correctly', () => {  

  const tree = renderer
    .create(
        <SetFeaturedWalkForm />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});