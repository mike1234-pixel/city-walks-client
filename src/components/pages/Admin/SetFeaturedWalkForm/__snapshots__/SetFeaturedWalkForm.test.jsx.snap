// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetFeaturedWalkForm Page renders correctly 1`] = `
<div>
  <h2>
    Set Featured Walks
  </h2>
  <p>
    Administrator note: a prerequisite of this working is that at least one record has the featuredWalk field set to true.
  </p>
  <p>
    Only records where walk is fully in Title Case can be set as a featured walk. For example, 'Bond Street to Baker Street' will not pass.
  </p>
  <p>
    The form will convert your input to Title Case.
  </p>
  <form
    className="set-featured-walk-form"
    onSubmit={[Function]}
  >
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="featured-walk-1"
        name="featured-walk-1"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="featured-walk-1"
        id=""
        onClick={[Function]}
      >
        featured walk 1 (walk name, e.g 'Stoke Newington')
      </label>
    </div>
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="featured-walk-2"
        name="featured-walk-2"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="featured-walk-2"
        id=""
        onClick={[Function]}
      >
        featured walk 2 (walk name, e.g 'Buckingham Palace')
      </label>
    </div>
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="featured-walk-3"
        name="featured-walk-3"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="featured-walk-3"
        id=""
        onClick={[Function]}
      >
        featured walk 3 (walk name, e.g 'Old Street')
      </label>
    </div>
    <button
      className="btn-elegant btn-outline-elegant btn Ripple-parent"
      data-test="button"
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="submit"
    >
      Set Featured Walks
      <i
        className="far fa-star ml-2"
        data-test="fa"
      />
      <div
        className="Ripple Ripple-outline "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
  </form>
</div>
`;
